apps/costcogas/costco_gas.star

```diff
@@ -0,0 +1,387 @@
+            value = "gas-icon",
+        ),
+    ]
+
+    price_colours = [
+        schema.Option(
+            display = "White",
+            value = "white",
+        ),
+        schema.Option(
+            display = "Red Gas, Green Diesel",
+            value = "red-green",
+        ),
+        schema.Option(
+            display = "Green Gas, Red Diesel",
+            value = "green-red",
+        ),
+    ]
+
+    time_formats = [
+        schema.Option(
+            display = "24 Hour Clock",
+            value = "24-hours",
+        ),
+        schema.Option(
+            display = "12 Hour Clock",
+            value = "12-hours",
+        ),
+    ]
+
+    return schema.Schema(
+        version = "1",
+        fields = [
+            schema.LocationBased(
+                id = "warehouse_by_loc",
+                name = "Warehouse",
+                desc = "A list of warehouses by location",
+                icon = "location-dot",
+                handler = get_warehouses,
+            ),
+            schema.Toggle(
+                id = "show_hours",
+                name = "Show Hours",
+                desc = "Show open hours for the current day. If enabled, icon won't show.",
+                icon = "business-time",
+                default = DEFAULT_CONFIG["show_hours"],
+            ),
+            schema.Dropdown(
+                id = "time_format",
+                name = "Time Format",
+                desc = "24 or 12 hour clock for hours display",
+                icon = "clock",
+                default = DEFAULT_CONFIG["time_format"],
+                options = time_formats,
+            ),
+            schema.Dropdown(
+                id = "icon_display",
+                name = "Icon",
+                desc = "Icon to display",
+                icon = "icons",
+                default = DEFAULT_CONFIG["icon_display"],
+                options = icons,
+            ),
+            schema.Dropdown(
+                id = "price_colour",
+                name = "Price Colour",
+                desc = "Colour scheme for price display",
+                icon = "palette",
+                default = DEFAULT_CONFIG["price_colour"],
+                options = price_colours,
+            ),
+        ],
+    )
+
+def get_cached_data(url, ttl):
+    cached_data = cache.get(url)
+
+    if cached_data != None:
+        response_data = json.decode(cached_data)
+    else:
+        http_data = http.get(url, headers = API_HEADERS)
+        if http_data.status_code != 200:
+            fail("HTTP request failed with status {} for URL {}".format(http_data.status_code, url))
+        response_data = [x for x in http_data.json() if type(x) == "dict"]
+        cache.set(url, json.encode(response_data), ttl_seconds = ttl)
+
+    return response_data
+
+def get_warehouses(location):
+    loc = json.decode(location)
+    warehouses = get_cached_data(API_WAREHOUSE_SEARCH.format("&latitude=" + loc["lat"] + "&longitude=" + loc["lng"]), 86400)
```

Comment:
Would you mind adding some code to truncate latitude and longitude? This would help protect against leaking users' exact location to the API provider. As a secondary benefit, it also improves cache rate a bit in densely populated areas.

I'm thinking 1 or 2 decimal points would be sufficient, based on [this discussion](https://gis.stackexchange.com/questions/8650/measuring-accuracy-of-latitude-and-longitude)?

You can find some examples over here: https://github.com/tidbyt/community/search?q=truncate

Reply:
Done! Truncated to one decimal point as that should be resonably precise for a list of stores